c++/src/kj/async-inl.h

```diff
@@ -351,8 +351,10 @@ class PromiseDisposer {
 
   static void dispose(PromiseArenaMember* node) {
     PromiseArena* arena = node->arena;
+    // Defer the `delete` to protect against exception in `destroy()`.
+    // Reminder: `delete` automatically ignores null pointers
+    KJ_DEFER(delete arena);
     node->destroy();
```

Comment:
Any way of having a test for this?

Reply:
as I wrote in comment - there is already a test for this. It is just that capnproto doesn't have an asan CI configuration.